algos/src/sample_rate_conversion.rst

```diff
@@ -349,94 +343,92 @@ the output is:
-resampler for 3/4 ratio that is used in e.g. 32 to 24 kHz conversion.
+optimization. The next figure shows derivation of the polyphase fractional
+resampler for a 3/4 ratio that is used in , for example, a 32 to 24 kHz conversion.
 
 .. figure:: poly34.png
 
    Polyphase decomposition for fractional 32 to 24 kHz conversion (3/4)
 
-1. the basic conversion is shown. 
+1. The basic conversion is shown.
 
 2. The interpolation is changed to polyphase filter where low-pass
-   filter H(z) is split to three sub-filters R\ :sub:`0`\(z),
+   filter H(z) is split into three sub-filters R\ :sub:`0`\(z),
    R\ :sub:`1`\(z), and R\ :sub:`2`\(z).
 
-3. The “3 to 1 commutator” structure the z\ :sup:`-1` unit delays are
+3. The “3 to 1 commutator” structure that the z\ :sup:`-1` unit delays are
    multiplicated to match the decimation rate of 4. The subfilter outputs
    need to be compensated with additional negative delay (zp, p > 0)
-   added to preserve the sub-filter out to whole filter chain output
-   Y(z).
+   to preserve the sub-filter out to whole filter chain output Y(z).
 
-4. The added negative delays are moved to filter input side by
-   dividing the negative delay by interpolation factor. Also the
-   decimation at filter output is moved to commutator input side.
+4. The added negative delays are moved to filter the input side by
+   dividing the negative delay by an interpolation factor. Also, the
+   decimation at filter output is moved to the commutator input side.
 
-5. The order of decimation and interpolation swapped to have
-   decimation first. Also a delay is added to input to compensate for
-   negative delay used to make the filter causal.
+5. The order of decimation and interpolation are swapped to have
+   decimation first. Also, a delay is added to the input to compensate for
+   a negative delay used to make the filter causal.
 
 6. The input side delays are merged.
 
 Note that the sub-filters R(z) in practical implementation share the
-same delay line. The delay length is defined the length of the longest
+same delay line. The delay length is defined as the length of the longest
 delay chain needed.
 
-Also in a practical implementation this delay length include the
+Also, in a practical implementation, this delay length includes the
 length of processing block length and store multiple channels of
 audio.
 
-In this example the output commutator after the reformatting remained
-unit delays based. In case of non-unit delays a more complex
+In this example the output commutator, after reformatting, remained
+unit delays-based. In case of non-unit delays, a more complex
 interleaving output buffer structure is needed.
 
-In the next example rate conversion up-samples with ratio of 3/2 for
-e.g. 32 kHz to 48 kHz conversion. The structure is the same for
-down-sampling conversion
+In the next example, rate conversion up-samples with a ratio of 3/2 such
+as 32 kHz to 48 kHz conversion. The structure is the same for
```

Comment:
My gibberish.. Would it be better as "In the next example of polyphase decomposition, the input is up-sampled by 3/2 e.g. 32 kHz to 38 kHz"?